src/tests/stub_libmem.cc

```diff
@@ -39,16 +39,12 @@ void *memAllocate(mem_type)
     return xmalloc(64*1024);
 }
 
-void *memAllocRigid(size_t net_size)
-{
-    return xmalloc(net_size);
-}
-
 void *
 memAllocBuf(size_t net_size, size_t * gross_size)
 {
-    *gross_size=net_size;
-    return xcalloc(1, net_size);
+    if (gross_size)
```

Comment:
If this PR effectively makes the undocumented memAllocBuf() gross_size address _optional_, it should make the corresponding change in memReallocBuf() as well, to keep the two sister functions in sync: All buffers allocated using memAllocBuf() ought to remain compatible with memReallocBuf().

Please do not forget to update _all_ memReallocBuf() implementations, including the one in src/mem/minimal.cc.

Reply:
It is **effectively** optional now, the callers have not changed their expectations nor requirements. Whenever it can be provided it should be provided.

The `memReallocBuf` suffers from the design flaw that looking up which pool is responsible for deallocating `oldbuf` requires knowing `oldbuf` size. That is passed using `gross_size` - so this parameter is **not** optional for `memReallocBuf()`.

Dropping `memReallocBuf()` entirely is in the planning for a followup PR as soon as these larger removals get out of the way.

Reply:
> `gross_size` is **not** optional for `memReallocBuf()`.

Agreed. Sorry I missed that fact when writing my change request.

> Dropping `memReallocBuf()` entirely is in the planning for a followup PR

I am resolving my change request under the assumption that you will promptly post a followup PR that removes memReallocBuf() and, hence, address the problem this change request was dedicated to -- a new dangerous inconsistency between memAllocBuf() and memReallocBuf().